api/src/shop/stripe_checkout.py

```diff
@@ -0,0 +1,252 @@
+            f"Attempting to start new subscription {subscription_type} using checkout session id {checkout_session_id}")
+        subscription_start = "now"
+
+        memberships = get_membership_summary(member_id)
+
+        if (memberships.membership_active and subscription_type == SubscriptionTypes.MEMBERSHIP):
+          subscription_start = calc_start_ts(memberships.membership_end)
+        elif (memberships.labaccess_active and subscription_type == SubscriptionTypes.LAB):
+          subscription_start = calc_start_ts(memberships.labaccess_end)
+
+        member: Member = db_session.query(Member).get(member_id)
+        stripe_customer = get_stripe_customer(member)
+
+        price_id = lookup_subscription_price_for(member, subscription_type)
+        if (price_id == None):
+          return {'error': f"Unable to find suitable subscription type {subscription_type} for {member}"}
+
+        active_subscriptions = get_stripe_subscriptions(stripe_customer['id'])
+        print(f"Checking if user has active subscription for price {price_id}")
+        for active_subscription in active_subscriptions:
+          for item in active_subscription['items']['data']:
+            print(f" - Found price id {item['price']['id']}")
+            if (item['price']['id'] == price_id):
+              # There is already an active subscription. Let's bail out
+              return {'error': "Member already has active subscription"}
+
+        metadata = {
+            MSMetaKeys.USER_ID: member_id,
+            MSMetaKeys.MEMBER_NUMBER: member.member_number,
+            "subscription_type": subscription_type,
+        }
+
+        subscription_schedule = stripe.SubscriptionSchedule.create(start_date=subscription_start,
+                                                                   customer=stripe_customer['id'],
+                                                                   phases=[
+                                                                       {
+                                                                           'items': [
+                                                                               {
+                                                                                   'price': price_id,
+                                                                                   'metadata': metadata,
+                                                                               },
+                                                                           ],
+                                                                           'collection_method': 'charge_automatically',
+                                                                           'metadata': metadata,
+                                                                           'proration_behavior': 'none',
+                                                                       }
+                                                                   ])
```

Comment:
Newline to reduce indent:

```suggestion
        subscription_schedule = stripe.SubscriptionSchedule.create(
            start_date=subscription_start,
            customer=stripe_customer['id'],
            phases=[
                {
                    'items': [
                        {
                            'price': price_id,
                            'metadata': metadata,
                        },
                    ],
                    'collection_method': 'charge_automatically',
                    'metadata': metadata,
                    'proration_behavior': 'none',
                }
            ])
```